node_modules/oae-content/lib/api.js

```diff
@@ -482,60 +483,73 @@ var _createContent = function(ctx, contentId, revisionId, resourceSubType, displ
             }
 
-            // Ensure the specified folders exist and can be managed by the current user
-            canManageFolders(ctx, folders, function(err, folders) {
+            principals = _.values(principals);
+
+            AuthzAPI.canInteract(ctx, ctx.tenant().alias, principals, function(err, canInteract, illegalPrincipalIds) {
                 if (err) {
                     return callback(err);
+                } else if (!_.isEmpty(illegalPrincipalIds)) {
+                    return callback({'code': 400, 'msg': 'One or more target members being granted access are not authorized to become members on this content item'});
                 }
 
-                ContentDAO.Content.createContent(contentId, revisionId, ctx.user().id, resourceSubType, displayName, description, visibility, otherValues, revisionData, function(err, contentObj, revision) {
+                // Ensure the specified folders exist and can be managed by the current user
+                canManageFolders(ctx, folders, function(err, folders) {
                     if (err) {
                         return callback(err);
                     }
 
-                    // Make the current user a manager of the content
-                    additionalMembers[ctx.user().id] = ContentConstants.roles.MANAGER;
-                    _setContentPermissions(ctx, contentObj, additionalMembers, function(err, contentObj) {
+                    ContentDAO.Content.createContent(contentId, revisionId, ctx.user().id, resourceSubType, displayName, description, visibility, otherValues, revisionData, function(err, contentObj, revision) {
                         if (err) {
                             return callback(err);
                         }
 
-                        var allMembers = _.union(principals, [ctx.user()]);
-                        ContentMembersLibrary.insert(contentObj, allMembers, function(err) {
+                        // Make the current user a manager of the content
+                        permissions[ctx.user().id] = ContentConstants.roles.MANAGER;
+                        _setContentPermissions(ctx, contentObj, permissions, function(err, contentObj) {
                             if (err) {
-                                log().warn({
-                                    'err': err,
-                                    'contentId': contentObj.id,
-                                    'principalIds': _.pluck(allMembers, 'id')
-                                }, 'An error occurred while inserting members into a content members library while creating a content item');
+                                return callback(err);
                             }
 
-                            // Add the content item to the specified folders, if any
-                            _addContentItemToFolders(ctx, contentObj, folders, function(err) {
+                            var allMembers = _.union(principals, [ctx.user()]);
```

Comment:
If the current user was already passed in, will we be inserting it into their library twice?